### ISSUE TITLE
`cabal new-build --dependencies-only` builds more than deps

### ISSUE THREAD

**Issue by hvr** [haskell/cabal#3490]
For instance, in `hackage-server`'s source-tree:

```
$ rm -rf dist-newstyle/

$ cabal new-build --dep -w ghc-7.10.3 --disable-tests --dry
In order, the following would be built (use -v for more details):
hackage-server-0.5.1

$ cabal new-build --dep -w ghc-7.10.3 --disable-tests 
Resolving dependencies...
In order, the following will be built (use -v for more details):
hackage-server-0.5.1
Configuring hackage-server-0.5.1...
Preprocessing executable 'hackage-build' for hackage-server-0.5.1...
[ 1 of 12] Compiling Distribution.Server.Framework.MemSize ( Distribution/Server/Framework/MemSize.hs, /stuff3/work/GitHub/hackage-server/dist-newstyle/build/hackage-server-0.5.1/build/hackage-build/hackage-build-tmp/Distribution/Server/Framework/MemSize.o )
[ 2 of 12] Compiling Distribution.Server.Framework.AuthTypes ( Distribution/Server/Framework/AuthTypes.hs, /stuff3/work/GitHub/hackage-server/dist-newstyle/build/hackage-server-0.5.1/build/hackage-build/hackage-build-tmp/Distribution/Server/Framework/AuthTypes.o )
[ 3 of 12] Compiling Distribution.Server.Util.GZip ( Distribution/Server/Util/GZip.hs, /stuff3/work/GitHub/hackage-server/dist-newstyle/build/hackage-server-0.5.1/build/hackage-build/hackage-build-tmp/Distribution/Server/Util/GZip.o )
...
```

I'd expect `--dep` to result in building all non-local packages required by the build-plan, and no local packages (in the case of building out of `Cabal`+`cabal-install`'s Git repo,  `hackage-security` would be a local package, as it depends on the local `Cabal` package from Git)

**Comment by dcoutts** [comment]
This should be just a matter of porting over the code for `--only-dependencies` in the existing install code path. It works just by pruning the install plan to remove the targets (and checking that that is sane).

**Comment by hvr** [comment]
Addressed by #3815 (once merged)

**Comment by 23Skidoo** [comment]
#3815 has been merged.
